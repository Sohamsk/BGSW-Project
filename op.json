{"Name":"whatever", "ModuleType": "something"body: [{"RuleType": "DeclareVariable""Identifier": "I","Type": "INTEGER"},{"RuleType": "DeclareVariable""Identifier": "A","Type": "VARIANT"},"LetStatement": {},"LetStatement": {},"LetStatement": {},]}