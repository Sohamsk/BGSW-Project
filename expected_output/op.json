{"FileName":"test4", "FileType": "bas","body": [{"RuleType": "DeclareVariable","Identifier": "counter","Type": "Integer"},{"RuleType": "expression", "Body": [{"Type":"ambiguousIdentifier", "Symbol":"counter"},{"Type":"Operator","Symbol": "="},{"Type":"integerLiteral", "Symbol":"1"}]},{"RuleType":"DoLoopStatement","Kind":"until","BeforeLoop":false,"Condition": [{"Type":"ambiguousIdentifier", "Symbol":"counter"},{"Type":"Operator","Symbol": ">"},{"Type":"integerLiteral", "Symbol":"5"}],"Body": [{"RuleType": "expression", "Body": [{"Type":"ambiguousIdentifier", "Symbol":"counter"},{"Type":"Operator","Symbol": "="},{"Type":"ambiguousIdentifier", "Symbol":"counter"},{"Type":"Operator","Symbol": "+"},{"Type":"integerLiteral", "Symbol":"1"}]}]}]}